### PR TITLE
refactor: cleanup artwork screen ff

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Clean up of the `ARArtworkRedesingPhase2` feature flag

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Clean up of the `ARArtworkRedesingPhase2` feature flag - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Artwork/Artwork.tests.tsx [hunk 1]
```diff
@@ -65,48 +61,12 @@ describe("Artwork", () => {
...
-    // ArtworkBelowTheFoldQuery
-    resolveMostRecentRelayOperation(environment)
-
-    await flushPromiseQueue()
-
-    expect(screen.UNSAFE_queryByType(ImageCarousel)).toBeTruthy()
-    expect(screen.UNSAFE_queryByType(CommercialInformation)).toBeTruthy()
-    expect(screen.UNSAFE_queryByType(ActivityIndicator)).toBeNull()
-    expect(screen.UNSAFE_queryByType(ArtworkDetails)).toBeTruthy()
-  })
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Artwork.tests.tsx:108]: **question**: can we update this test cases for redesigned artwork page?

File: src/app/Scenes/Artwork/Artwork.tests.tsx [hunk 2]
```diff
@@ -418,18 +368,13 @@ describe("Artwork", () => {
...
-        expect(screen.queryByText("00d  00h  00m  00s")).toBeTruthy()
       })
     })
   })
 
   describe("Partner Section", () => {
-    it("should not display partner link if CBN flag is on", () => {
-      __globalStoreTestUtils__?.injectFeatureFlags({ AREnableConversationalBuyNow: true })
-
+    it("should not display partner link", () => {
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Artwork.tests.tsx:377]: **question**: should we check case/some render result when `AREnableConversationalBuyNow` feature flag is enabled?

File: src/app/Scenes/Artwork/Artwork.tsx [hunk 3]
```diff
@@ -798,7 +585,6 @@ export const ArtworkContainer = createRefetchContainer(
     `,
     me: graphql`
       fragment Artwork_me on Me {
-        ...CommercialInformation_me
         ...ArtworkStickyBottomContent_me
       }
     `,
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Artwork.tsx:590]: **suggestion**: please remove `isSold` field [here](https://github.com/artsy/eigen/blob/c5197dcb6cf3f2bcd14bbcadae1a36803b04118f/src/app/Scenes/Artwork/Artwork.tsx#L496)

Comment (dimatretyak) [reply src/app/Scenes/Artwork/Artwork.tsx:590]: `id` field [here](https://github.com/artsy/eigen/blob/c5197dcb6cf3f2bcd14bbcadae1a36803b04118f/src/app/Scenes/Artwork/Artwork.tsx#L491)

Comment (dimatretyak) [reply src/app/Scenes/Artwork/Artwork.tsx:590]: `artists` field [here](https://github.com/artsy/eigen/blob/c5197dcb6cf3f2bcd14bbcadae1a36803b04118f/src/app/Scenes/Artwork/Artwork.tsx#L499-L501)

Comment (dimatretyak) [reply src/app/Scenes/Artwork/Artwork.tsx:590]: `partner.id` field [here](https://github.com/artsy/eigen/blob/c5197dcb6cf3f2bcd14bbcadae1a36803b04118f/src/app/Scenes/Artwork/Artwork.tsx#L526)

Comment (dimatretyak) [reply src/app/Scenes/Artwork/Artwork.tsx:590]: `sale.id` field [here](https://github.com/artsy/eigen/blob/c5197dcb6cf3f2bcd14bbcadae1a36803b04118f/src/app/Scenes/Artwork/Artwork.tsx#L538)

**Review by dimatretyak — DISMISSED** [review]
Great cleanup 👌

**Comment by dimatretyak** [comment]
also please don't forget to deprecate feature flag in Echo 🙏

**Comment by gkartalis** [comment]
done [here](https://github.com/artsy/echo/pull/416)

**Comment by damassi** [comment]
Screenshot 2023-02-21 at 8 47 22 AM

🔥

File: src/app/Scenes/Artwork/Artwork.tests.tsx (hunk 2 above)
Comment (gkartalis) [reply src/app/Scenes/Artwork/Artwork.tests.tsx:377]: will also add a case for this one

File: src/app/Scenes/Artwork/Artwork.tsx (hunk 3 above)
Comment (gkartalis) [reply src/app/Scenes/Artwork/Artwork.tsx:590]: thanks! Will address all of those

**Review by dblandin — DISMISSED** [review]
🙌🏼

**Review by dimatretyak — APPROVED** [review]
Thank you for all addressed changes

**Pull request merged**